notebooks/rocoto.ipynb

```diff
@@ -0,0 +1,1013 @@
+    "<!--cell 22-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "33c67471-84a1-4b0f-b0bd-40f805e6615f",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "workflow:\n",
+      "  attrs:\n",
+      "    realtime: false\n",
+      "    scheduler: slurm\n",
+      "  cycledef:\n",
+      "    - spec: 202410290000 202410300000 06:00:00\n",
+      "  log: logs/test.log\n",
+      "  tasks:\n",
+      "    task_bacon:\n",
+      "      command: \"echo Cooking bacon...\"\n",
+      "      cores: 1\n",
+      "      walltime: 00:00:10\n",
+      "    task_eggs:\n",
+      "      command: \"echo Cooking eggs...\"\n",
+      "      nodes: 1:ppn=4\n",
+      "      walltime: 00:00:10\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/rocoto/tasks-workflow.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "08e9c593-7ba1-45da-8f1a-99389c9dfdf5",
+   "metadata": {},
+   "source": [
+    "Each task may optionally have one or more dependencies that must be accounted for before a task runs. These are specified using a `dependency:` block within the `task_*` block that the dependencies apply to. Dependencies are structured as boolean expressions using a variety of keys that may define specific types of dependencies like task or data dependencies. They may also group dependencies together using boolean operators keys like `and` or `or`. For a full list of possible tags, see the <a href=\"https://christopherwharrop.github.io/rocoto/\">Rocoto Documentation</a>. Below, the `task_eggs:` block includes one data dependency indicated by the `datadep` key, plus a `value` key that identifies the required data. The `task_serve:` block includes two task dependencies for the bacon and eggs tasks. Since there are multiple dependencies here, they need to be contained within a boolean operator block that describes how to deal with the group of dependencies which may not all have the same level of completion. Here the `and:` block indicates that all of the individual tasks (i.e. `task_eggs`) within need to be completed. The two task dependencies must have unique names since they exist at the same level, and they are differentiated here using the `_name` suffix. To prevent circular dependencies, task dependencies must have a `task` attribute that indicates the name of a task that is already defined above it. Similar to the `workflow:` block, an `attrs:` block is used here to add attributes to `taskdep`, and the `task` key specifies the value of the task attribute.\n",
```

Comment:
I'm not sure how to include the suggestion here in GH, but I think a new paragraph would be good starting with "Below, the `task_eggs:` block includes..."